### PR TITLE
ref: Replace organization id type with newtype

### LINKED ISSUE #4069: Introduce new-type for `OrganizationId`
We already have a new-types for a few things, like the `ProjectId`, we should also have one for the organization id. Some crates already define type aliases `type OrganizationId = u64`, all of these should be replaced with a proper new-type.

### PR DESCRIPTION
**Pull request by Litarnus**
This PR introduces a `OrganizationId` new-type struct which replaces all old occurrences of `u64` where it was applicable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,7 @@
 - Feature flags of graduated features are now hard-coded in Relay so they can be removed from Sentry. ([#4076](https://github.com/getsentry/relay/pull/4076), [#4080](https://github.com/getsentry/relay/pull/4080))
 - Add parallelization in Redis commands. ([#4118](https://github.com/getsentry/relay/pull/4118))
 - Extract user ip for spans. ([#4144](https://github.com/getsentry/relay/pull/4144))
+- Replace u64 with OrganizationId new-type struct for organization id. ([#4159](https://github.com/getsentry/relay/pull/4159))
```
Comment (iambriccardo) [inline CHANGELOG.md]:
```suggestion
- Replace u64 with `OrganizationId` new-type struct for organization id. ([#4159](https://github.com/getsentry/relay/pull/4159))
```

**Review by loewenheim — APPROVED** [review]

**Review by iambriccardo — APPROVED** [review]
LGTM, did you make sure serialization and deserialization is the same as before?

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,6 +32,7 @@
 - Feature flags of graduated features are now hard-coded in Relay so they can be removed from Sentry. ([#4076](https://github.com/getsentry/relay/pull/4076), [#4080](https://github.com/getsentry/relay/pull/4080))
 - Add parallelization in Redis commands. ([#4118](https://github.com/getsentry/relay/pull/4118))
 - Extract user ip for spans. ([#4144](https://github.com/getsentry/relay/pull/4144))
+- Replace u64 with `OrganizationId` new-type struct for organization id. ([#4159](https://github.com/getsentry/relay/pull/4159))
```
Comment (Dav1dde) [inline CHANGELOG.md]: Careful this is the changelog from the previous version, when a new release is built, we need to manually recreate the section, it's called `## Unreleased` until the release process renames it to the next proper release.

File: relay-base-schema/src/organization.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+//! Contains [`OrganizationId`] which is the ID of a Sentry organization and is currently a
+//! a wrapper over `u64`.
+
+use serde::{Deserialize, Serialize};
+
+/// The unique identifier of a Sentry organization
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+pub struct OrganizationId(u64);
```
Comment (Dav1dde) [inline relay-base-schema/src/organization.rs:8]: Can add a unit test to make sure serialization is correct using `serde_json`.

File: relay-base-schema/src/organization.rs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+//! Contains [`OrganizationId`] which is the ID of a Sentry organization and is currently a
+//! a wrapper over `u64`.
+
+use serde::{Deserialize, Serialize};
+
+/// The unique identifier of a Sentry organization
```
Comment (Dav1dde) [inline relay-base-schema/src/organization.rs]:
```suggestion
/// The unique identifier of a Sentry organization.
```

File: relay-base-schema/src/organization.rs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+pub struct OrganizationId(u64);
+
+impl OrganizationId {
+    /// Creates a new organization ID from its numeric value
+    #[inline]
+    pub fn new(id: u64) -> Self {
+        OrganizationId(id)
```
Comment (Dav1dde) [inline relay-base-schema/src/organization.rs]:
```suggestion
        Self(id)
```

File: relay-base-schema/src/organization.rs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    /// returns the numeric value of the organization ID
```
Comment (Dav1dde) [inline relay-base-schema/src/organization.rs]:
```suggestion
    /// Returns the numeric value of the organization ID.
```

**Review by Dav1dde — APPROVED** [review]

File: relay-base-schema/src/organization.rs (hunk 3 above)
Comment (Litarnus) [reply relay-base-schema/src/organization.rs:8]: I will add one. Just for curiosity, do you know if it's currently being serialized to JSON somewhere? I didn't see anything

**Comment by Litarnus** [comment]
> LGTM, did you make sure serialization and deserialization is the same as before?

I've added tests to make sure it de/serialized the same as u64

File: relay-base-schema/src/organization.rs (hunk 3 above)
Comment (Dav1dde) [reply relay-base-schema/src/organization.rs:8]: In the `ProjectInfo` struct, that is de- and serialized from JSON.

**Comment by Dav1dde** [comment]
@Litarnus 👀

**Pull request merged**